bundles/org.eclipse.swt/Eclipse SWT/win32/org/eclipse/swt/widgets/MessageBox.java

```diff
@@ -99,6 +106,88 @@ public MessageBox (Shell parent) {
 public MessageBox (Shell parent, int style) {
 	super (parent, checkStyle (parent, checkStyle (style)));
 	checkSubclass ();
+	/* Create the callback hook */
+	int threadId = OS.GetCurrentThreadId ();
+	Callback cbtCallback = new Callback (this, "CBTProc", 3); //$NON-NLS-1$
```

Comment:
This is a leak, and a pretty bad one, because the limit for Callbacks is quite low (see `MAX_CALLBACKS`)

Reply:
After `UnhookWindowsHookEx()` called `cbtCallback.dispose()` to avoid this leak.